programs/associated-token-account/instructions.go

```diff
@@ -67,7 +69,11 @@ func (inst *Instruction) Accounts() (out []*solana.AccountMeta) {
 }
 
 func (inst *Instruction) Data() ([]byte, error) {
-	return []byte{}, nil
+	buf := new(bytes.Buffer)
+	if err := ag_binary.NewBinEncoder(buf).Encode(inst); err != nil {
+		return nil, fmt.Errorf("unable to encode instruction: %w", err)
+	}
+	return buf.Bytes(), nil
```

Comment:
This is copy and pate code to match the implementation of `Data` of the other instruction types which include data. This basically just calls the `MarshalWithEncoder` method defined above